### ISSUE TITLE
Error on webpack run

### ISSUE THREAD

**Issue by Pavel-Durov** [dumconstantin/ramda-loader#10]
Following README Installation instructions I've encountered this error when running webpack:

```
Version: webpack 3.8.1
Time: 459ms
    Asset    Size  Chunks             Chunk Names
bundle.js  2.7 kB       0  [emitted]  app
   [0] .../app.js 230 bytes {0} [built] [failed] [1 error]

ERROR in .../app.js
Module build failed: TypeError: Cannot add property debug, object is not extensible
    at Object.module.exports (/.../node_modules/ramda-loader/dist/loader.js:39:14)
```

**Comment by dumconstantin** [comment]
Hi,

It seems to be something regarding the webpack version.
Can you try some of the fixes described here? https://github.com/preboot/angular-webpack/issues/239

**Comment by Pavel-Durov** [comment]
Hi, 

I can see that this discussion is about beta version of webpack : "2.1.0-beta.25" (about year ago)

I'm currently running latest stable 3.8.1 version. 
I tried switching versions and I was able to run successfully on 1.5.3.

Was able to reproduse in the following versions: 3.8.0, 3.8.1, 3.7.1, 3.7.0,  3.6.0, 3.5.6, 3.5.5.
I'll try to see if I can resolve it with webpack configuration tweaking :)

**Comment by dumconstantin** [comment]
The problem originates here:
https://github.com/dumconstantin/ramda-loader/blob/d01c288695a842be8ea8209c12b425c6d22bd9e5/src/loader.js#L14

If you can clone the query (e.g. cloneDeep from Lodash, or Ramda.clone) it should fix the problem.
Although it would need a bit of testing on webpack 2,3 (maybe added in the test suite).

I'll be very happy to merge a pull request if you'd like to have a go at it.

**Comment by Pavel-Durov** [comment]
Yes that would be great. I'll work on it, thanks.

**Comment by Pavel-Durov** [comment]
It seems like the problem is deeper than that.

I did some debugging, and from what I can see - **self** is a non-extendable object. 
Described here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cant_define_property_object_not_extensible

first exception line:
https://github.com/dumconstantin/ramda-loader/blob/d01c288695a842be8ea8209c12b425c6d22bd9e5/src/loader.js#L17

same issue with other **self** property assignments.

Exception:
> Module build failed: TypeError: Cannot add property **debug**, object is **not extensible**

I don't know what exactly caused this change in **module.export** function (webpack version probably), but I guess **self** object needs to be changed.

I tried to use  Object.assign and it worked (although it's sort of a hack).
changed:
https://github.com/dumconstantin/ramda-loader/blob/d01c288695a842be8ea8209c12b425c6d22bd9e5/src/loader.js#L11
to:
```
var self = Object.assign({}, this)
```
Tested with README.md examples, but **npm test** failed :(

What do you think?

**Comment by dumconstantin** [comment]
Hi,

From what I've seen in the Webpack 3 docs, the plugin needs a bit of refactor to match the new way for doing things.

I'm very tight on schedule atm and won't be able to handle it... In essence is just taking a boilerplate Webpack 3 plugin and adding in the logic from the current ` ramda-loader ` implementation...

**Comment by Pavel-Durov** [comment]
Thanks! I guess we can close this ticket then :)

**Comment by dumconstantin** [comment]
Published under 0.5.2.

Great work Pavel!
